x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/popover_editor.tsx

```diff
@@ -56,6 +56,31 @@ function asOperationOptions(operationTypes: OperationType[], compatibleWithCurre
     }));
 }
 
+const LabelInput = ({ value, onChange }: { value: string; onChange: (value: string) => void }) => {
+  const [inputValue, setInputValue] = useState(value);
+
+  useEffect(() => {
+    setInputValue(value);
+  }, [value, setInputValue]);
+
+  const onChangeDebounced = _.debounce(onChange, 256);
```

Comment:
This is on me - because of the `useEffect` this breaks now because the debounced change from 256ms ago overwrites what the user has typed till then:
![overlaps](https://user-images.githubusercontent.com/1508364/85988204-fdd26300-b9ee-11ea-9fe3-412db6d9729e.gif)

The fix is really easy though - this only happens because the calls are not actually debounced, they are just delayed by 256 ms. `debounce` keeps internal state about a call already "in flight". As it is called in the component body here, there is a new internal state per typed key, so it's not actually debouncing

The fix for this is really simple:

```
const onChangeDebounced = useMemo(() => _.debounce(onChange, 256), [onChange]);
```

By memoizing the debounce call, it will hold the same internal state for each invocation, thus only firing the state update by the end of a series of keystrokes.